File: app/src/main/java/com/fphoenixcorneae/viewpagertransformer/ZoomInTransformer.java
package com.fphoenixcorneae.viewpagertransformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * Created on 2017/6/30.
 */

public class ZoomInTransformer extends BaseTransformer {
    @Override
    protected void onTransform(View page, float position) {
        final float scale = position < 0 ? position + 1F : Math.abs(1F - position);
        ViewHelper.setScaleX(page, scale);
        ViewHelper.setScaleY(page, scale);
        ViewHelper.setPivotX(page, page.getWidth() * 0.5F);
        ViewHelper.setPivotY(page, page.getHeight() * 0.5F);
        ViewHelper.setAlpha(page, position < -1F || position > 1F ? 0F : 1F - (scale - 1F));
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/view_pager_transformer/ZoomInTransformer.kt
package com.fphoenixcorneae.view_pager_transformer

import android.view.View
import com.nineoldandroids.view.ViewHelper
import kotlin.math.abs

/**
 * @desc 放大的转换效果
 * @date 2017-06-30
 */
class ZoomInTransformer : AbstractBaseTransformer() {
    override fun onTransform(page: View, position: Float) {
        val scale = if (position < 0) position + 1f else abs(1f - position)
        ViewHelper.setScaleX(page, scale)
        ViewHelper.setScaleY(page, scale)
        ViewHelper.setPivotX(page, page.width * 0.5f)
        ViewHelper.setPivotY(page, page.height * 0.5f)
        ViewHelper.setAlpha(page, if (position < -1f || position > 1f) 0f else 1f - (scale - 1f))
    }
}